components/scream/src/share/io/output_manager.hpp

```diff
@@ -0,0 +1,271 @@
+#ifndef SCREAM_OUTPUT_MANAGER_HPP
+#define SCREAM_OUTPUT_MANAGER_HPP
+
+#include "share/io/scorpio_output.hpp"
+#include "share/io/scream_scorpio_interface.hpp"
+
+#include "ekat/mpi/ekat_comm.hpp"
+#include "ekat/ekat_parameter_list.hpp"
+#include "share/field/field_repository.hpp"
+#include "share/grid/grids_manager.hpp"
+
+namespace scream
+{
+
+/*
+ * The OutputManager Class handles all individual output streams.
+ * Rather than have the SCREAM-AD or unit tests call the scorpio_output
+ * class objects directly, the OutputManager stores these in a vector,
+ * m_output_streams.
+ *
+ * Similar to a typical atmospheric process object, the OutputManager
+ * has an init, run and finalize routine which is called during the AD
+ * during those respective steps.
+ *
+ * init():
```

Comment:
Consider moving method descriptions so they are above their methods. This might encourage people to document their own code similarly. :-)